packages/plugin-electron-preload-error/preload-error.js

```diff
@@ -27,6 +27,8 @@ module.exports = app => ({
 
         event.context = context
 
+        event._isPreloadError = true
```

Comment:
This is untidy. An alternative would be to add an `onerror` callback to the `client._notify()` call below, but this would run _after_ user defined callbacks, so the value wouldn't be correct for those callbacks.

Reply:
Maybe it'd be a bit cleaner if this set the process type directly and the process info plugin avoided overwriting it, e.g.

```js
// plugin-preload-error
event.addMetadata('process', { type: 'preload' })

// plugin-process-info
info.type = event.getMetadata('process', 'type') || 'something else'
```

Reply:
I did something similar to this. It actually runs twice (once in main and once in the renderer) so it needs to be a bit smarter about what it overwrites, but I got it working.